src/Action_Hbond.cpp

```diff
@@ -76,6 +80,9 @@ Action::RetType Action_Hbond::Init(ArgList& actionArgs, ActionInit& init, int de
   useAtomNum_ = actionArgs.hasKey("printatomnum");
   acut_ = actionArgs.getKeyDouble("angle",135.0);
   noIntramol_ = actionArgs.hasKey("nointramol");
+  if (actionArgs.hasKey("atomindex")) {
+    AHDIndices_ = (DataSet_integer*) init.DSL().AddSet(DataSet::INTEGER, "atomindex");
```

Comment:
I forgot to mention this before; the reason you get this error is you are setting up your DataSet too early. The `hbsetname_` variable isn't assigned until after all keywords and masks are processed; this is also a convention (and for a wonder is actually mentioned in the dev. guide in the `Adding Actions` section). So, you should set up your DataSet after the comment `// Setup datasets`. Use a `bool` to save whether your keyword is there or not:
```C++
bool useAtomIndex = actionArgs.hasKey("atomindex");
```
then probably after the `if (calcSolvent_)` section, put
```C++
if (useAtomIndex)
  AHDIndices_ = (DataSet_integer*)init.DSL().AddSet(DataSet::INTEGER, MetaData(hbsetname_,"atomindex"));
```
Make sense?